diesel_derives/tests/selectable.rs

```diff
@@ -114,3 +114,28 @@ fn embedded_option_with_nullable_field() {
     let data = my_structs::table.select(A::as_select()).get_result(conn);
     assert!(data.is_err());
 }
+
+#[test]
+fn manually_specified_expression() {
+    table! {
+        my_structs (foo) {
+            foo -> Integer,
+            bar -> Nullable<Text>,
+        }
+    }
+
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Queryable, Selectable)]
+    #[diesel(table_name = my_structs)]
+    struct A {
+        foo: i32,
+        #[diesel(
+            select_expression = my_structs::bar.is_not_null(),
+            select_expression_type = dsl::IsNotNull<my_structs::bar>
+        )]
+        bar_is_set: bool,
+    }
+
+    let conn = &mut connection();
+    let data = my_structs::table.select(A::as_select()).get_result(conn);
```

Comment:
I don't understand the SQL query that's generated here. Is it something like `SELECT bar IS NOT NULL FROM my_structs;`?

Reply:
`SELECT foo, bar IS NOT NULL FROM my_structs`, as foo is also present in the struct.

Reply:
Oh, so we are getting boolean instead of bar value. Makes sense.